pkg/virt-controller/network/pod_annotations.go

```diff
@@ -0,0 +1,33 @@
+package network
+
+import (
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	virtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/network/deviceinfo"
+	"kubevirt.io/kubevirt/pkg/network/downwardapi"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
+func GeneratePodAnnotations(networks []virtv1.Network, interfaces []virtv1.Interface, multusStatusAnnotation string, bindingPlugins map[string]virtv1.InterfaceBindingPlugin) map[string]string {
+	newAnnotations := map[string]string{}
+	if vmispec.SRIOVInterfaceExist(interfaces) {
+		networkPCIMapAnnotationValue := deviceinfo.CreateNetworkPCIAnnotationValue(
+			networks, interfaces, multusStatusAnnotation,
+		)
+		newAnnotations[deviceinfo.NetworkPCIMapAnnot] = networkPCIMapAnnotationValue
+	}
+	if vmispec.BindingPluginNetworkWithDeviceInfoExist(interfaces, bindingPlugins) {
+		networkDeviceInfoMap, err := deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, interfaces, multusStatusAnnotation, bindingPlugins)
```

Comment:
I think this is the only production usage of this function, therefore it may be a good idea to reconsider its signature.
There is duplication with `BindingPluginNetworkWithDeviceInfoExist` and I hope we can avoid adding that other function.

How about this: The `bindingPlugins` is only needed for the filtering, both here and in the prev one.
So we can pass in an already filtered interfaces slice.
```go
ifacesWithDeviceInfo := vmispec.FilterInterfacesSpec(interfaces, func(iface v1.Interface) bool {
    if iface.Binding != nil {
        if binding, exist := bindingPlugins[iface.Binding.Name]; exist && binding.DownwardAPI == v1.DeviceInfo {
            return true
        }
    }
    return false
})
if len(ifacesWithDeviceInfo) > 0 {
    networkDeviceInfoMap, err := deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, ifacesWithDeviceInfo, multusStatusAnnotation)
    if err != nil {
        ...
    }
    ...
}
return newAnnotations
```

Reply:
IMO the only readability problem is with `binding, exist := bindingPlugins[iface.Binding.Name]; exist && binding.DownwardAPI == v1.DeviceInfo` , which you could extract to a function if you want.

If the problem is re-use, you can also just move the filter to `vmispec` and name it.
```go
vmispec.FilterInterfacesSpec(interfaces, func(iface v1.Interface) bool {
    if iface.Binding != nil {
        if binding, exist := bindingPlugins[iface.Binding.Name]; exist && binding.DownwardAPI == v1.DeviceInfo {
            return true
        }
    }
    return false
})
```

We will benefit here from the filtered interfaces slice.

I find the current helper in `vmispec` not elegant.

But if you do not like it, I won't insist further.